### PR TITLE
ah-gawb-1396: Library: autocomplete during metadata intake 

### PR DESCRIPTION
**Pull request by ahaessly**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs [hunk 1]
```diff
@@ -131,6 +131,27 @@
...
+                      :cache false
+                      :prepare (fn [query settings]
+                                 (clj->js
+                                   (assoc (js->clj settings)
+                                     :headers {:Authorization (str "Bearer " (utils/get-access-token))}
+                                     :type "GET"
+                                     :url (str (aget settings "url") query)
+                                     )))}
+     :typeaheadDisplay (fn [result]
+                         (.text (js/$ (str "<div>" result "</div>"))))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs]: It's really suspicious to me that we're taking a thing, wrapping it in an element, then getting the text out.  I'm not sure what's going on under the covers, but I understand why you needed to do this, and we should investigate simplifying this.

File: src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs [hunk 2]
```diff
@@ -131,6 +131,27 @@
...
+     :placeholder (:inputHint prop)
+     :bloodhoundInfo {:url (str (config/api-url-root) "/api/library/populate/suggest/" (name property) "/")
+                      :cache false
+                      :prepare (fn [query settings]
+                                 (clj->js
+                                   (assoc (js->clj settings)
+                                     :headers {:Authorization (str "Bearer " (utils/get-access-token))}
+                                     :type "GET"
+                                     :url (str (aget settings "url") query)
+                                     )))}
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs]: put end braces on previous line

File: src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs [hunk 3]
```diff
@@ -131,6 +131,27 @@
                              :onClick #(apply swap! state update :attributes dissoc
                                               (library-utils/get-related-id+label-props library-schema property))})]]))])
 
+(defn- render-populate-typeahead [{:keys [prop property]}]
+  [:div {:style {:marginBottom "0.75em"}}
+   [comps/AutocompleteFilter
+    {:width "100%"
+     :ref "text-filter"
+     :placeholder (:inputHint prop)
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs]: `prop` itself is merged into the data handed to each of these render functions, so you can include it in the map destructure at the top (i.e. `[{:keys [prop property inputHint]}]`) and use it directly here.  Then again, I see that I myself didn't do that below..... so either leave it or change both.

File: src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs [hunk 4]
```diff
@@ -206,6 +227,7 @@
                 (cond enum (render-enum data)
                       (= type "boolean") (render-boolean data)
                       (= (:datatype renderHint) "freetext") (render-freetext data)
-                      (= (:typeahead prop) "ontology") (render-typeahead data)
+                      (= (:typeahead prop) "ontology") (render-ontology-typeahead data)
+                      (= (:typeahead prop) "populate") (do (utils/log property) (render-populate-typeahead data))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs]: Remove log statement

**Review by MattPutnam — APPROVED** [review]
👍 with minor cleanup/formatting, plus maybe file a story to take a look at the gymnastics we have to do in `:typeaheadDisplay`.

**Review by davidangb — APPROVED** [review]
code looks good; did not test locally.

**Pull request merged**